x-pack/plugins/siem/server/plugin.ts

```diff
@@ -52,10 +48,8 @@ export interface SetupPlugins {
   ml?: MlSetup;
 }
 
-export interface StartPlugins {
-  actions: ActionsStart;
-  alerting: AlertingStart;
-}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface StartPlugins {}
 
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface PluginSetup {}
```

Comment:
You can optionally remove these two empty interfaces if you want and just do this below:

```ts
export class Plugin implements IPlugin<void, void, SetupPlugins, StartPlugins> {
```

And then remove the `return {}` you have in the `start` and `setup` to not return anything unless I'm missing something and we are returning empty objects for a purpose. But it looks like the contracts allow void for return types and we don't have anything useful to return at the moment.

Reply:
@FrankHassanabad we return `{}` instead of `undefined` to allow consumers to distinguish between a disabled plugin and an enabled plugin with no contract. We first encountered this with our newsfeed integration, which resulted in https://github.com/elastic/kibana/pull/56236.